apps/ejabberd/src/ejabberd_hooks.erl

```diff
@@ -111,7 +111,7 @@ run(Hook, Host, Args) ->
     case ets:lookup(hooks, {Hook, Host}) of
         [{_, Ls}] ->
             mongoose_metrics:increment_generic_hook_metric(Host, Hook),
-            run1(Ls, Hook, Args);
+            run1(Ls, Hook, #{}, Args); % run with empty map as accumulator
```

Comment:
Would it be possible to remove the `run1` completely and call `run_fold1` instead of `run1`?